internal/federation/server.go

```diff
@@ -339,7 +339,7 @@ func (s *Server) MustCreateEvent(t *testing.T, room *ServerRoom, ev b.Event) *go
 
 // MustJoinRoom will make the server send a make_join and a send_join to join a room
 // It returns the resultant room.
-func (s *Server) MustJoinRoom(t *testing.T, deployment *docker.Deployment, remoteServer gomatrixserverlib.ServerName, roomID string, userID string) *ServerRoom {
+func (s *Server) MustJoinRoom(t *testing.T, deployment *docker.Deployment, remoteServer gomatrixserverlib.ServerName, roomID string, userID string, partialState ...bool) *ServerRoom {
```

Comment:
I guess this is go's idiom for an optional argument? Feels a bit odd to allow an arbitrary length array/slice, but I guess it's easier than updating every call site?

Reply:
Not sure actually. Functional varargs (https://stackoverflow.com/a/26326418) is the pattern [MustDoFunc uses](https://github.com/matrix-org/complement/blob/5c444e7d440feb4bde44f332fd0df8f67ddb6f3d/internal/client/client.go#L584). Here I didn't bother with them since there's only one optional argument.